src/pybind/mgr/cephadm/module.py

```diff
@@ -2107,27 +2119,59 @@ def _find_inv_for_host(hostname: str, inventory_dict: dict):
                     return _inventory
             raise OrchestratorError("No inventory found for host: {}".format(hostname))
 
-        ret = []
-        # 3) iterate over matching_host and call DriveSelection and to_ceph_volume
+        # 3) iterate over matching_host and call DriveSelection
         self.log.debug(f"Checking matching hosts -> {matching_hosts}")
         for host in matching_hosts:
             inventory_for_host = _find_inv_for_host(host, self.cache.devices)
             self.log.debug(f"Found inventory for host {inventory_for_host}")
-            drive_selection = selector.DriveSelection(drive_group, inventory_for_host)
+            drive_selection = DriveSelection(drive_group, inventory_for_host)
             self.log.debug(f"Found drive selection {drive_selection}")
-            cmd = translate.to_ceph_volume(drive_group, drive_selection).run()
-            self.log.debug(f"translated to cmd {cmd}")
-            if not cmd:
-                self.log.debug("No data_devices, skipping DriveGroup: {}".format(drive_group.service_name()))
-                continue
-            self.log.info('Applying %s on host %s...' % (
-                drive_group.service_name(), host))
-            ret_msg = self._create_osd(host, cmd)
-            ret.append(ret_msg)
-        return ", ".join(ret)
+            host_ds_map.append((host, drive_selection))
+        return host_ds_map
 
-    def _create_osd(self, host, cmd):
+    def set_unmanaged_flag(self, service_id, unmanaged_flag):
+        specs = self.spec_store.find(service_name=service_id)
```

Comment:
* `daemon_id` is something like `rgw.zone-us1.myhost.xxyyzz`
* `service_name` is something like `rgw.zone-us1`
* `service_id` is something like `zone-us1`.

thus `service_name` can never equal a `servcie_id`